src/parser/core/modules/PetTimeline.tsx

```diff
@@ -0,0 +1,87 @@
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Timeline, {Group, Item} from 'parser/core/modules/Timeline'
+import React from 'react'
+
+// This module puts pet skills on the timeline beneath the GCD spells
+export default class PetTimeline extends Module {
+	static handle = 'pettimeline'
+
+	@dependency private timeline!: Timeline
+
+	private petContainerGroupId = 'pet'
+	private petAutoGroupId = 'petauto'
+	private petCommandGroupId = 'petcommand'
+	private autoCasts: CastEvent[] = []
+	private commandCasts: CastEvent[] = []
+
+	protected canPetBeCommanded() {
+		return false
+	}
+
+	protected isCommandedEvent(event: CastEvent) {
+		return false
+	}
+
+	protected init() {
+		this.addEventHook('cast', {by: 'pet'}, this.onCast)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent) {
+		if (this.isCommandedEvent(event)) {
+			this.commandCasts.push(event)
+		} else {
+			this.autoCasts.push(event)
+		}
+	}
+
+	private onComplete() {
+		if (this.canPetBeCommanded()) {
+			this.timeline.addGroup(new Group({
+				id: this.petContainerGroupId,
+				content: 'Pet',
```

Comment:
Could the row name maybe be configurable? `Pet` works for SMN and SCH but sounds a bit weird for MCH or DRK for example.